cns/requestcontroller/kubecontroller/crdrequestcontroller.go

```diff
@@ -199,7 +200,9 @@ func (crdRC *crdRequestController) initCNS() error {
 
 	// Convert to CreateNetworkContainerRequest if crd not nill and is populated
 	if nodeNetConfig != nil && len(nodeNetConfig.Status.NetworkContainers) != 0 {
-		if ncRequest, err = CRDStatusToNCRequest(&nodeNetConfig.Status); err != nil {
+		// Assign the reference to the ncRequest, otherwise we'll pass a nil pointer to InitCNSState
```

Comment:
Its okay to pass the nil pointer. Actually it should be nil pointer in case CRD doesnt exist. Otherwise we wont be able to differiante between a bad NCRequest vs no ncrequest

Reply:
Right, that's why I had to add the ncRequestReference variable, see the comment below

Reply:
Nvm, saw your other comment, making a pointer